_security/configuration/configuration.md

```diff
@@ -139,6 +139,83 @@ If `challenge` is set to `true`, the Security plugin sends a response with statu
+
+You have the option to configure the Security plugin for username rate limiting, IP address rate limiting, or both. These configurations are made in the `config.yml` file. See the following sections for information about each type of rate limiting configuration.
+
+
+### Username rate limiting
+
+This configuration limits login attempts by username. When a login fails, the username is blocked for any machine in the network. The following example shows `config.yml` file settings configured for username rate limiting:
+
+```yml
+auth_failure_listeners:
+      internal_authentication_backend_limiting:
+        type: username
+        authentication_backend: internal
+        allowed_tries: 3
+        time_window_seconds: 60
+        block_expiry_seconds: 60
+        max_blocked_clients: 100000
+        max_tracked_clients: 100000
+```
+{% include copy.html %}
+
+The following table describes the individual settings for this type of configuration.
+
+| Setting | Description |
+| :--- | :--- |
+| `type` |  The type of rate limiting. In this case, `username`. |
+| `authentication_backend` | The internal backend. Enter `internal`. |
+| `allowed_tries` |  The number of login attempts allowed before login is blocked. Be aware that increasing the number increases heap usage. |
+| `time_window_seconds` | The window of time in which the value for `allowed_tries` is enforced. For example, if `allowed_tries` is `3` and `time_window_seconds` is `60`, a username has three attempts to log in successfully within a 60-second time span before login is blocked. |
+| `block_expiry_seconds` | The duration of time that login remains blocked after a failed login. After this time elapses, login is reset and the username can attempt successful login again. |
+| `max_blocked_clients` |  The maximum number of blocked usernames. This limits heap usage to avoid a potential DoS. |
+| `max_tracked_clients` | The maximum number of tracked usernames that have failed login. This limits heap usage to avoid a potential DoS. |
+
+
+### IP address rate limiting
+
+This configuration limits login attempts by IP address. When a login fails, the IP address specific to the machine being used for login is blocked. 
+
+There are two steps for configuring IP address rate limiting. First, set the `challenge` setting to `false` in the `http_authenticator` section of the `config.yml` file.
+
+```yml
+http_authenticator:
+  type: basic
+  challenge: false
+```
+
+For more information about this setting, see [HTTP basic authentication](/security/configuration/configuration/#http-basic-authentication).
+
+Second, configure the IP address rate limiting settings. The following example shows a completed configuration:
```

Comment:
```suggestion
Second, configure the IP address's rate limiting settings. The following example shows a completed configuration:
```

Reply:
Will get an editorial opinion on this. They aren't really settings for the IP address. The settings are for the feature called "IP address rate limiting". 
Maybe I should call it "IP client rate limiting". That was another alternative from the dev (both were supposed to be accurate). Would that sound better, read better, make more sense when reading it? Thanks.

Reply:
Since we're using the naming convention "IP address rate limiting," leave as originally written. The term is a compound noun. Alternatively, you could use the setting name: "Second, configure the `ip_rate_limiting` setting."